app/navigation/ReasonNavigator.scala

```diff
@@ -54,6 +54,12 @@ class ReasonNavigator @Inject()() {
     
     case AdviceProfessionPage => _ => routes.AdviceGivenController.onPageLoad(NormalMode)
 
+    case AdviceGivenPage => ua => ua.get(AdviceGivenPage) match {
+      case (Some(value)) if value.contactPreference == AdviceContactPreference.Email => routes.WhichEmailAddressCanWeContactYouWithController.onPageLoad(NormalMode)
+      case (Some(value)) if value.contactPreference == AdviceContactPreference.Telephone => routes.CanWeUseTelephoneNumberToContactYouController.onPageLoad(NormalMode)
+      case _ => routes.WhatIsTheReasonForMakingADisclosureNowController.onPageLoad(NormalMode)
```

Comment:
This needs to be the CheckYourAnswersController

Reply:
Fixed